src/Model.php

```diff
@@ -371,7 +373,39 @@ public function init()
             $this->addField($this->id_field, [
                 'system' => true,
             ]);
+        } else {
+            return; // don't declare actions for model without id_field
         }
+
+        if ($this->read_only) {
+            return; // don't declare action for read-only model
+        }
+
+        // Declare our basic CRUD actions for the model.
+        $this->addAction('add', [
+            'fields'  => true,
+            'scope'   => UserAction\Generic::NO_RECORDS,
+            'callback'=> 'save',
+            'ui'      => ['icon'=>'plus'],
+        ]);
+        $this->addAction('edit', [
+            'fields'  => true,
+            'scope'   => UserAction\Generic::SINGLE_RECORD,
+            'callback'=> 'save',
+            'ui'      => ['icon'=>'edit'],
+        ]);
+        $this->addAction('delete', [
+            'scope'=> UserAction\Generic::SINGLE_RECORD,
+            'ui'   => ['icon'=>'trash', 'danger', 'confirm'=>'Are you sure?'],
+        ]);
+
+        $this->addAction('validate', [
+            //'scope'=> any!
```

Comment:
Probably should be UserAction\Generic::SINGLE_RECORD scope, but also could be UserAction\Generic::NO_RECORDS. UserAction\Generic do not support multiple scopes for one action. Maybe it should?

Reply:
Yes I was considering. Will discuss tomorrow